.github/workflows/_win-test.yml

```diff
@@ -27,11 +27,33 @@ env:
   GIT_DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
 
 jobs:
+  # This needs to be run right before the test starts so that it can gather the
+  # latest labels from the PR
+  filter:
+    runs-on: [self-hosted, linux.large]
+    outputs:
+      test-matrix: ${{ steps.filter.outputs.test-matrix }}
+      is-test-matrix-empty: ${{ steps.filter.outputs.is-test-matrix-empty }}
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
```

Comment:
Well, we could consider tag.  It would make rolling out new GH action safer.  The only reason is that we don't support it for any action at the moment because these actions don't change that frequently, but it's a mechanism we can think about in the future.